artemis/modules/wordpress_plugins.py

```diff
@@ -0,0 +1,300 @@
+    "customizer-export-import",
+    "disable-xml-rpc-api",
+    "zarinpal-woocommerce-payment-gateway",
+    "metricool",
+    "button-contact-vr",
+    "visual-footer-credit-remover",
+    "sumome",
+]
+PLUGINS_TO_SKIP_CHANGELOG = ["wp-members"]
+PLUGINS_TO_SKIP_STABLE_TAG = ["scheduled-post-trigger", "pdf-viewer-for-elementor"]
+
+
+def get_version_from_readme(slug: str, readme_content: str) -> Optional[str]:
+    previous_line = ""
+    changelog_version = None
+
+    # These plugins' changelogs are reversed
+    if slug in PLUGINS_WITH_REVERSED_CHANGELOGS:
+        has_reversed_changelog = True
+    else:
+        has_reversed_changelog = False
+
+    if slug not in PLUGINS_TO_SKIP_CHANGELOG:
+        seen_changelog_line = False
+        for line in readme_content.lower().split("\n"):
+            line = line.strip("= #[]\r\t")
+            if not line:
+                continue
+
+            # Happens between changelog header and version, let's skip
+            if line.startswith("for the plugin's full changelog") or line.startswith("this changelog is for"):
+                continue
+
+            if previous_line == "changelog" or (has_reversed_changelog and seen_changelog_line):
+                seen_changelog_line = True
+                # Some changelog entries have the format <slug>: <version>
+                if line.startswith(slug):
+                    line = line[len(slug) :].strip(" :")
+                # Some changelog entries have the format version <version>
+                if "version" in line:
+                    line = line[line.find("version") + len("version") :].strip(" :")
+                # Some changelog entries have the format V <version>
+                if "v " in line:
+                    line = line[line.find("v ") + len("v ") :].strip(" :")
+                version = (
+                    line.replace("(", " ")
+                    .replace("*", " ")
+                    .replace("[", " ")
+                    .replace("]", " ")
+                    .replace("'", " ")
+                    .replace(":", " ")
+                    .replace(",", " ")
+                    .replace("-", " ")
+                    .replace("<h4>", " ")
+                    .replace("</h4>", " ")
+                    .strip()
+                    # Some versions are prefixed with 'v' (e.g. v1.0.0)
+                    .lstrip("v")
+                    .split(" ")[0]
+                )
+                if "." in version:
+                    changelog_version = version
+                    if not has_reversed_changelog:
+                        break
+
+            previous_line = line
+
+    # Some plugins have broken "stable tag"
+    if slug in PLUGINS_TO_SKIP_STABLE_TAG:
+        return changelog_version
+
+    tag_lines = [line for line in readme_content.lower().split("\n") if line.strip("* ").startswith("stable tag:")]
+    if len(tag_lines) > 1:
+        return changelog_version
+
+    if len(tag_lines):
```

Comment:
I don't get this, before this if statement you check if there are at least 2 tag_lines, then you again check for length of tag_lines. Should not it be just `if tag_lines` or `if len(tag_lines) == 1`?